### PR TITLE
Implement localisation support

### PR DESCRIPTION
**Pull request by youtux**
Fixes https://github.com/pytest-dev/pytest-bdd/issues/397
Supersedes https://github.com/pytest-dev/pytest-bdd/pull/345

Checklist:
- [x] Find all usages of `keyword` and check if we need to replace them with `keyword_type`

### CODE REVIEW

**Comment by youtux** [comment]
@jsa34 mind reviewing this too?

**Comment by jsa34** [comment]
There are some places where keywords are hardcoded, but I needed to resolve these for the rules PR, so this should fix this issue so that the gherkin terminal reporter and cucumber json out is using the desired keywords from the feature file, rather than assuming

**Comment by jsa34** [comment]
We should include the language in the cucumber json output.

https://github.com/cucumber/compatibility-kit/blob/main/devkit/samples/minimal/minimal.feature
https://github.com/cucumber/compatibility-kit/blob/main/devkit/samples/minimal/minimal.feature.ndjson

This is example of a minimal out that is cucumber-spec-compliant, and as no language is specified, it defaults to "en" in this example, but obviously we should pass the value coming from the gherkin parser.

**Review by jsa34 — CHANGES_REQUESTED** [review]
See comments in the MR

**Comment by jsa34** [comment]
Also resolves #179

**Comment by jsa34** [comment]
I also forgot an example and change log entry: it would be helpful to demonstrate how to specify the language in the feature file

File: src/pytest_bdd/types.py
Comment (jsa34) [inline src/pytest_bdd/types.py:1]: There is another type to map: Conjunction (And, But and *)

Comment (youtux) [reply src/pytest_bdd/types.py:1]: I know, but that's just a "tokenizer" type, not a semantic one, that's why I didn't include it here

**Comment by youtux** [comment]
> We should include the language in the cucumber json output.
> 
> https://github.com/cucumber/compatibility-kit/blob/main/devkit/samples/minimal/minimal.feature https://github.com/cucumber/compatibility-kit/blob/main/devkit/samples/minimal/minimal.feature.ndjson
> 
> This is example of a minimal out that is cucumber-spec-compliant, and as no language is specified, it defaults to "en" in this example, but obviously we should pass the value coming from the gherkin parser.

Do you know if there is a specification somewhere for Cucumber JSON output?

**Comment by youtux** [comment]
> I also forgot an example and change log entry: it would be helpful to demonstrate how to specify the language in the feature file

yes I forgot indeed, added in 342cf5edcfb516d4a4136ff6b7e722e7e48590b7

**Comment by jsa34** [comment]
I think the parser test needs simplifying otherwise it will be a maintenance nightmare.
Perhaps just fix for this and then I will raise a PR to make the test more focussed.

**Comment by youtux** [comment]
> I think the parser test needs simplifying otherwise it will be a maintenance nightmare. Perhaps just fix for this and then I will raise a PR to make the test more focussed.

Agree, but I think that once gherkin-official v30.0.0 is released, we can remove most of the classes in `gherkin_parser.py`, so let's wait for that one.

Can you check this again?

**Review by jsa34 — APPROVED** [review]

**Comment by jsa34** [comment]
All good to me now :)

**Pull request merged**